From 20b5baf3e8075e5c1f62e6cc6f57e89212773928 Mon Sep 17 00:00:00 2001
From: Stefano Rivera <stefano@rivera.za.net>
Date: Tue, 8 Apr 2025 15:49:35 -0400
Subject: [PATCH] Some manpage cleanups

Submitted in https://bugs.debian.org/1101406 as the result of running a
linter on the manpage.

There is no content-change here, only some syntax improvements and
tweaks to formatting.

Forwarded: https://github.com/python/cpython/pull/132291
Bug-Debian: https://bugs.debian.org/1101406
---
 Misc/python.man | 154 +++++++++++++++++++++++++-----------------------
 1 file changed, 79 insertions(+), 75 deletions(-)

--- a/Misc/python.man
+++ b/Misc/python.man
@@ -1,7 +1,7 @@
-.TH PYTHON "1"
+.TH PYTHON 1
 
 .\" To view this file while editing, run it through groff:
-.\"   groff -Tascii -man python.man | less
+.\"   groff -Tascii -man -ww python.man |& less
 
 .SH NAME
 python \- an interpreted, interactive, object-oriented programming language
@@ -80,7 +80,7 @@
 ]
 .br
        [
-.B \--check-hash-based-pycs
+.B \-\-check-hash-based-pycs
 .I default
 |
 .I always
@@ -89,16 +89,16 @@
 ]
 .br
        [
-.B \--help
+.B \-\-help
 ]
 [
-.B \--help-env
+.B \-\-help\-env
 ]
 [
-.B \--help-xoptions
+.B \-\-help\-xoptions
 ]
 [
-.B \--help-all
+.B \-\-help\-all
 ]
 .br
        [
@@ -144,14 +144,14 @@
 .TP
 .B \-b
 Issue warnings about str(bytes_instance), str(bytearray_instance)
-and comparing bytes/bytearray with str. (-bb: issue errors)
+and comparing bytes/bytearray with str. (\-bb: issue errors)
 .TP
 .BI "\-c " command
 Specify the command to execute (see next section).
 This terminates the option list (following options are passed as
 arguments to the command).
 .TP
-.BI "\-\-check-hash-based-pycs " mode
+.BI "\-\-check\-hash\-based\-pycs " mode
 Configure how Python evaluates the up-to-dateness of hash-based .pyc files.
 .TP
 .B \-d
@@ -162,17 +162,16 @@
 Ignore environment variables like PYTHONPATH and PYTHONHOME that modify
 the behavior of the interpreter.
 .TP
-.B \-h ", " \-? ", "\-\-help
+.BR \-h ", " \-? ", " \-\-help
 Prints the usage for the interpreter executable and exits.
 .TP
-.B "\-\-help\-env"
+.B \-\-help\-env
 Prints help about Python-specific environment variables and exits.
 .TP
-.B "\-\-help\-xoptions"
+.B \-\-help\-xoptions
 Prints help about implementation-specific \fB\-X\fP options and exits.
 .TP
-.TP
-.B "\-\-help\-all"
+.B \-\-help\-all
 Prints complete usage information and exits.
 .TP
 .B \-i
@@ -183,9 +182,11 @@
 raises an exception.
 .TP
 .B \-I
-Run Python in isolated mode. This also implies \fB\-E\fP, \fB\-P\fP and \fB\-s\fP. In
-isolated mode sys.path contains neither the script's directory nor the user's
-site-packages directory. All PYTHON* environment variables are ignored, too.
+Run Python in isolated mode.
+This also implies \fB\-E\fP, \fB\-P\fP and \fB\-s\fP.
+In isolated mode sys.path contains neither the script's directory nor the
+user's site\-packages directory.
+All PYTHON* environment variables are ignored, too.
 Further restrictions may be imposed to prevent the user from injecting
 malicious code.
 .TP
@@ -203,13 +204,13 @@
 adding .opt-1 before the .pyc extension.
 .TP
 .B \-OO
-Do \fB-O\fP and also discard docstrings; change the filename for
+Do \fB\-O\fP and also discard docstrings; change the filename for
 compiled (bytecode) files by adding .opt-2 before the .pyc extension.
 .TP
 .B \-P
 Don't automatically prepend a potentially unsafe path to \fBsys.path\fP such
-as the current directory, the script's directory or an empty string. See also the
-\fBPYTHONSAFEPATH\fP environment variable.
+as the current directory, the script's directory or an empty string.
+See also the \fBPYTHONSAFEPATH\fP environment variable.
 .TP
 .B \-q
 Do not print the version and copyright messages. These messages are
@@ -243,7 +244,7 @@
 searching for a module.  Also provides information on module cleanup
 at exit.
 .TP
-.B \-V ", " \-\-version
+.BR \-V ", " \-\-version
 Prints the Python version number of the executable and exits.  When given
 twice, print more information about the build.
 
@@ -256,26 +257,28 @@
 The simplest settings apply a particular action unconditionally to all warnings
 emitted by a process (even those that are otherwise ignored by default):
 
-  -Wdefault  # Warn once per call location
-  -Werror    # Convert to exceptions
-  -Walways   # Warn every time
-  -Wall      # Same as -Walways
-  -Wmodule   # Warn once per calling module
-  -Wonce     # Warn once per Python process
-  -Wignore   # Never warn
+.nf
+  \-Wdefault  # Warn once per call location
+  \-Werror    # Convert to exceptions
+  \-Walways   # Warn every time
+  \-Wall      # Same as \-Walways
+  \-Wmodule   # Warn once per calling module
+  \-Wonce     # Warn once per Python process
+  \-Wignore   # Never warn
+.fi
 
 The action names can be abbreviated as desired and the interpreter will resolve
 them to the appropriate action name. For example,
-.B -Wi
+.B \-Wi
 is the same as
-.B -Wignore .
+.BR \-Wignore .
 
 The full form of argument is:
-.IB action:message:category:module:lineno
+.IB action : message : category : module : lineno
 
 Empty fields match all values; trailing empty fields may be omitted. For
 example
-.B -W ignore::DeprecationWarning
+.B \-W ignore::DeprecationWarning
 ignores all DeprecationWarning warnings.
 
 The
@@ -304,10 +307,10 @@
 equivalent to an omitted line number.
 
 Multiple
-.B -W
+.B \-W
 options can be given; when a warning matches more than one option, the action
 for the last matching option is performed. Invalid
-.B -W
+.B \-W
 options are ignored (though, a warning message is printed about invalid options
 when the first warning is issued).
 
@@ -320,16 +323,16 @@
 .TP
 .BI "\-X " option
 Set implementation-specific option. The following options are available:
+.nf
+    \fB\-X cpu_count=\fIN\fR: override the return value of \fIos.cpu_count()\fR;
+       \fB\-X cpu_count=default\fR cancels overriding; also \fBPYTHON_CPU_COUNT\fI
 
-    \fB-X cpu_count=\fIN\fR: override the return value of \fIos.cpu_count()\fR;
-       \fB-X cpu_count=default\fR cancels overriding; also \fBPYTHON_CPU_COUNT\fI
-
-    \fB-X dev\fR: enable CPython's "development mode", introducing additional
+    \fB\-X dev\fR: enable CPython's "development mode", introducing additional
         runtime checks which are too expensive to be enabled by default. It
         will not be more verbose than the default if the code is correct: new
         warnings are only emitted when an issue is detected. Effect of the
         developer mode:
-           * Add default warning filter, as \fB-W default\fR
+           * Add default warning filter, as \fB\-W default\fR
            * Install debug hooks on memory allocators: see the
              PyMem_SetupDebugHooks() C function
            * Enable the faulthandler module to dump the Python traceback on a
@@ -338,59 +341,60 @@
            * Set the dev_mode attribute of sys.flags to True
            * io.IOBase destructor logs close() exceptions
 
-    \fB-X importtime\fR: show how long each import takes. It shows module name,
+    \fB\-X importtime\fR: show how long each import takes. It shows module name,
         cumulative time (including nested imports) and self time (excluding
         nested imports). Note that its output may be broken in multi-threaded
         application. Typical usage is
-        \fBpython3 -X importtime -c 'import asyncio'\fR
+        \fBpython3 \-X importtime \-c 'import asyncio'\fR
 
-    \fB-X faulthandler\fR: enable faulthandler
+    \fB\-X faulthandler\fR: enable faulthandler
 
-    \fB-X frozen_modules=\fR[\fBon\fR|\fBoff\fR]: whether or not frozen modules
+    \fB\-X frozen_modules=\fR[\fBon\fR|\fBoff\fR]: whether or not frozen modules
        should be used.
        The default is "on" (or "off" if you are running a local build).
 
-    \fB-X gil=\fR[\fB0\fR|\fB1\fR]: enable (1) or disable (0) the GIL; also
+    \fB\-X gil=\fR[\fB0\fR|\fB1\fR]: enable (1) or disable (0) the GIL; also
        \fBPYTHON_GIL\fR
-       Only available in builds configured with \fB--disable-gil\fR.
+       Only available in builds configured with \fB\-\-disable\-gil\fR.
 
-    \fB-X int_max_str_digits=\fInumber\fR: limit the size of int<->str conversions.
+    \fB\-X int_max_str_digits=\fInumber\fR: limit the size of int<->str conversions.
        This helps avoid denial of service attacks when parsing untrusted data.
        The default is sys.int_info.default_max_str_digits.  0 disables.
 
-    \fB-X no_debug_ranges\fR: disable the inclusion of the tables mapping extra
+    \fB\-X no_debug_ranges\fR: disable the inclusion of the tables mapping extra
        location information (end line, start column offset and end column
        offset) to every instruction in code objects. This is useful when
        smaller code objects and pyc files are desired as well as suppressing
        the extra visual location indicators when the interpreter displays
        tracebacks.
 
-    \fB-X perf\fR: support the Linux "perf" profiler; also \fBPYTHONPERFSUPPORT=1\fR
+    \fB\-X perf\fR: support the Linux "perf" profiler; also \fBPYTHONPERFSUPPORT=1\fR
 
-    \fB-X perf_jit\fR: support the Linux "perf" profiler with DWARF support;
+    \fB\-X perf_jit\fR: support the Linux "perf" profiler with DWARF support;
        also \fBPYTHON_PERF_JIT_SUPPORT=1\fR
 
-    \fB-X presite=\fIMOD\fR: import this module before site; also \fBPYTHON_PRESITE\fR
+    \fB\-X presite=\fIMOD\fR: import this module before site; also \fBPYTHON_PRESITE\fR
        This only works on debug builds.
 
-    \fB-X pycache_prefix=\fIPATH\fR: enable writing .pyc files to a parallel
+    \fB\-X pycache_prefix=\fIPATH\fR: enable writing .pyc files to a parallel
        tree rooted at the given directory instead of to the code tree.
 
-    \fB-X showrefcount\fR: output the total reference count and number of used
+    \fB\-X showrefcount\fR: output the total reference count and number of used
         memory blocks when the program finishes or after each statement in the
         interactive interpreter. This only works on debug builds
 
-    \fB-X tracemalloc\fR: start tracing Python memory allocations using the
+    \fB\-X tracemalloc\fR: start tracing Python memory allocations using the
         tracemalloc module. By default, only the most recent frame is stored in a
-        traceback of a trace. Use -X tracemalloc=NFRAME to start tracing with a
+        traceback of a trace. Use \-X tracemalloc=NFRAME to start tracing with a
         traceback limit of NFRAME frames
 
-    \fB-X utf8\fR: enable UTF-8 mode for operating system interfaces,
-        overriding the default locale-aware mode. \fB-X utf8=0\fR explicitly
+    \fB\-X utf8\fR: enable UTF-8 mode for operating system interfaces,
+        overriding the default locale-aware mode. \fB\-X utf8=0\fR explicitly
         disables UTF-8 mode (even when it would otherwise activate
         automatically). See \fBPYTHONUTF8\fR for more details
 
-    \fB-X warn_default_encoding\fR: enable opt-in EncodingWarning for 'encoding=None'
+    \fB\-X warn_default_encoding\fR: enable opt-in EncodingWarning for 'encoding=None'
+.fi
 
 .TP
 .B \-x
@@ -430,7 +434,7 @@
 is used,
 .I sys.argv[0]
 contains the string
-.I '-c'.
+.RI ' \-c '.
 Note that options interpreted by the Python interpreter itself
 are not placed in
 .IR sys.argv .
@@ -504,16 +508,16 @@
 specifying \fB\-d\fP multiple times.
 .IP PYTHONEXECUTABLE
 If this environment variable is set,
-.IB sys.argv[0]
+.IB sys.argv [0]
 will be set to its value instead of the value got through the C runtime. Only
 works on Mac OS X.
 .IP PYTHONFAULTHANDLER
 If this environment variable is set to a non-empty string,
-.IR faulthandler.enable()
+.IR faulthandler.enable ()
 is called at startup: install a handler for SIGSEGV, SIGFPE, SIGABRT, SIGBUS
 and SIGILL signals to dump the Python traceback.
 .IP
-This is equivalent to the \fB-X faulthandler\fP option.
+This is equivalent to the \fB\-X faulthandler\fP option.
 .IP PYTHON_FROZEN_MODULES
 If this variable is set to \fBon\fR or \fBoff\fR, it determines whether or not
 frozen modules are ignored by the import machinery.  A value of \fBon\fR means
@@ -524,12 +528,12 @@
 .IP PYTHON_GIL
 If this variable is set to 1, the global interpreter lock (GIL) will be forced
 on. Setting it to 0 forces the GIL off. Only available in builds configured
-with \fB--disable-gil\fP.
+with \fB\-\-disable\-gil\fP.
 .IP PYTHON_HISTORY
 This environment variable can be used to set the location of a history file
 (on Unix, it is \fI~/.python_history\fP by default).
 .IP
-This is equivalent to the \fB-X gil\fR option.
+This is equivalent to the \fB\-X gil\fR option.
 .IP PYTHONNODEBUGRANGES
 If this variable is set, it disables the inclusion of the tables mapping
 extra location information (end line, start column offset and end column
@@ -557,8 +561,8 @@
 interpreter itself, or to allow a cluster of python processes to share hash
 values.
 
-The integer must be a decimal number in the range [0,4294967295].  Specifying
-the value 0 will disable hash randomization.
+The integer must be a decimal number in the range [0,4\|294\|967\|295].
+Specifying the value 0 will disable hash randomization.
 .IP PYTHONHOME
 Change the location of the standard Python libraries.  By default, the
 libraries are searched in ${prefix}/lib/python<version> and
@@ -576,20 +580,20 @@
 A value of 0 disables the limit.  Conversions to or from bases 2, 4, 8,
 16, and 32 are never limited.
 .IP
-This is equivalent to the \fB-X int_max_str_digits=\fINUMBER\fR option.
+This is equivalent to the \fB\-X int_max_str_digits=\fINUMBER\fR option.
 .IP PYTHONIOENCODING
 If this is set before running the interpreter, it overrides the encoding used
 for stdin/stdout/stderr, in the syntax
-.IB encodingname ":" errorhandler
+.IB encodingname : errorhandler
 The
-.IB errorhandler
+.I errorhandler
 part is optional and has the same meaning as in str.encode. For stderr, the
-.IB errorhandler
+.I errorhandler
 part is ignored; the handler will always be \'backslashreplace\'.
 .IP PYTHONMALLOC
 Set the Python memory allocators and/or install debug hooks. The available
 memory allocators are
-.IR malloc
+.I malloc
 and
 .IR pymalloc .
 The available debug hooks are
@@ -599,7 +603,7 @@
 .IR pymalloc_debug .
 .IP
 When Python is compiled in debug mode, the default is
-.IR pymalloc_debug
+.I pymalloc_debug
 and the debug hooks are automatically used. Otherwise, the default is
 .IR pymalloc .
 .IP PYTHONMALLOCSTATS
@@ -679,14 +683,14 @@
 .IP
 The value of the variable is the maximum number of frames stored in a
 traceback of a trace. For example,
-.IB PYTHONTRACEMALLOC=1
+.B PYTHONTRACEMALLOC=1
 stores only the most recent frame.
 .IP PYTHONUNBUFFERED
 If this is set to a non-empty string it is equivalent to specifying
 the \fB\-u\fP option.
 .IP PYTHONUSERBASE
 Defines the user base directory, which is used to compute the path of the user
-.IR site-packages
+.I site\-packages
 directory and installation paths for
 .IR "python \-m pip install \-\-user" .
 .IP PYTHONUTF8
@@ -704,7 +708,7 @@
 .SS Debug-mode variables
 Setting these variables only has an effect in a debug build of Python, that is,
 if Python was configured with the
-\fB\--with-pydebug\fP build option.
+\fB\-\-with\-pydebug\fP build option.
 .IP PYTHONDUMPREFS
 If this environment variable is set, Python will dump objects and reference
 counts still alive after shutting down the interpreter.
@@ -718,7 +722,7 @@
 executed, and before the \fI__main__\fR module is created.
 This only works on debug builds.
 .IP
-This is equivalent to the \fB-X presite=\fImodule\fR option.
+This is equivalent to the \fB\-X presite=\fImodule\fR option.
 .SH AUTHOR
 The Python Software Foundation: https://www.python.org/psf/
 .SH INTERNET RESOURCES
